x-pack/elastic-agent/pkg/agent/install/install_unix.go

```diff
@@ -7,8 +7,59 @@
 
 package install
 
+import (
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/paths"
+)
+
 // postInstall performs post installation for unix-based systems.
 func postInstall() error {
 	// do nothing
 	return nil
 }
+
+func checkPackageInstall() bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+
+	// NOTE searching for english words might not be a great idea as far as portability goes.
+	// list all installed packages then search for paths.BinaryName?
+	// dpkg is strange as the remove and purge processes leads to the package bing isted after a remove, but not after a purge
+
+	// check debian based systems (or systems that use dpkg)
+	// If the package has been installed, the status starts with "install"
+	// If the package has been removed (but not pruged) status starts with "deinstall"
+	// If purged or never installed, rc is 1
+	if _, err := os.Stat("/etc/dpkg"); err == nil {
```

Comment:
What do you thing about doing a `which dpkg-query` instead? Only thing would be is `which` always present on a system? I believe so.